fuzz.go

```diff
@@ -179,6 +192,12 @@ const (
 )
 
 func (f *Fuzzer) doFuzz(v reflect.Value, flags uint64) {
+	if f.curDepth >= f.maxDepth {
+		return
+	}
+	f.curDepth++
```

Comment:
I would probably make another struct containing curDepth and the *Fuzzer to use, construct this object in the Fuzz method, and implement doFuzz on this new object.